server/functional-test/src/test/java/org/zanata/feature/editor/EditorFilterMessagesTest.kt

```diff
@@ -0,0 +1,85 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this software; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA, or see the FSF
+ * site: http://www.fsf.org.
+ */
+package org.zanata.feature.editor
+
+import org.junit.Before
+import org.junit.Test
+import org.junit.experimental.categories.Category
+import org.zanata.feature.Trace
+import org.zanata.feature.testharness.TestPlan.DetailedTest
+import org.zanata.feature.testharness.ZanataTestCase
+import org.zanata.page.webtrans.EditorPage
+import org.zanata.util.ZanataRestCaller
+import org.zanata.workflow.BasicWorkFlow
+import org.zanata.workflow.LoginWorkFlow
+
+import org.assertj.core.api.Assertions.assertThat
+import org.zanata.util.ZanataRestCaller.buildSourceResource
+import org.zanata.util.ZanataRestCaller.buildTextFlow
+
+/**
+ * @author Patrick Huang [pahuang@redhat.com](mailto:pahuang@redhat.com)
+ */
+@Category(DetailedTest::class)
+class EditorFilterMessagesTest : ZanataTestCase() {
+    private val document = "messages"
+
+    @Before
+    fun setUp() {
+        val restCaller = ZanataRestCaller()
+        val sourceResource = buildSourceResource(document,
+                buildTextFlow("res1", "hello world"),
+                buildTextFlow("res2", "greetings"),
+                buildTextFlow("res3", "hey"))
+        restCaller.postSourceDocResource("about-fedora", "master",
+                sourceResource, false)
+        LoginWorkFlow().signIn("admin", "admin")
+    }
+
+    @Trace(summary = "The user can filter translation entries using more " +
+            "than one search term")
+    @Test(timeout = MAX_SHORT_TEST_DURATION.toLong())
+    fun canFilterByMultipleFields() {
+        val editorPage = BasicWorkFlow()
+                .goToEditor("about-fedora", "master", "fr", document)
+
+        assertThat(editorPage.messageSources).containsExactly(
+                "hello world", "greetings", "hey")
+
+        val page = editorPage.inputFilterQuery("resource-id:res2")
+        page.waitForPageSilence()
+        assertThat(page.messageSources).contains("greetings")
+    }
+
+    @Trace(summary = "The user may save the filter url for later use")
+    @Test(timeout = MAX_SHORT_TEST_DURATION.toLong())
+    fun editorFilterIsBookmarkable() {
+        val urlForEditor = String.format(BasicWorkFlow.EDITOR_TEMPLATE, "about-fedora",
+                "master", "fr", document)
+        val urlWithFilterCondition = "$urlForEditor;search:hello%20w;resid:res1"
+        val editorPage = BasicWorkFlow().goToPage(urlWithFilterCondition,
+                EditorPage::class.java)
+
+        assertThat(editorPage.messageSources).containsExactly("hello world")
+        assertThat(editorPage.filterQuery.trim { it <= ' ' })
```

Comment:
trim what?

Reply:
It actually trims the filter query, which adds a space on the end - but I've changed it to test against the _actual_ string. Good catch.

Reply:
I'm not sure I really caught anything. I was trying to work out what exactly it was doing. Was it trimming any characters "less than or equal to" SPACE, ie control characters?